store/postgres/tests/chain_head.rs

```diff
@@ -0,0 +1,290 @@
+//! Test ChainStore implementation of Store, in particular, how
+//! the chain head pointer gets updated in various situations
+#[macro_use]
+extern crate diesel;
+
+use diesel::prelude::*;
+use diesel::RunQueryDsl;
+use futures::future::{self, IntoFuture};
+use lazy_static::lazy_static;
+use std::fmt::Debug;
+use std::sync::Arc;
+
+use graph::components::store::ChainStore;
+use graph::serde_json;
+use graph_store_postgres::Store as DieselStore;
+
+use test_store::*;
+
+// The ancestor count we use for chain head updates. We keep this very small
+// to make setting up the tests easier
+const ANCESTOR_COUNT: u64 = 3;
+
+// This is duplicated from `src/db_schema.rs` since there is no good way
+// to keep db_schema private, but allow accessing it from integration tests
```

Comment:
maybe something like:
```
#[cfg(debug_assertions)]
pub mod pub_db_schema {
   use db_schema::*;
}

Reply:
I added a commit to that effect - it's not perfect, in that `cargo test --release` will fail to compile, but I can't figure out a way to have `cargo` set a custom feature flag when it's compiling tests.

Reply:
Cool, yhea it's a hack but I don't think anyone runs `test --release`.